topostats/scars.py

```diff
@@ -0,0 +1,470 @@
+        self.max_scar_width = max_scar_width
+        self.min_scar_length = min_scar_length
+
+    def mark_if_positive_scar(
+        self,
+        row_col: tuple,
+        stddev: float,
+        img: np.ndarray,
+        marked: np.ndarray,
+        threshold_low: float,
+        max_scar_width: int,
+    ) -> None:
+        """
+        Determine if the points below and including the pixel at the specified row and column are an positive scar.
+        If they are, mark them in the marked 2d np.ndarray. Note that this only detects positive scars.
+        Parameters
+        ----------
+        row_col: tuple
+            A tuple containing the row and column indices of the pixel for the top of the potential scar. Note that
+            the first value is the row index, and the second is the column index.
+        stddev: float
+            The standard deviation, or the root-mean-square value for the image.
+        img: np.ndarray
+            A 2-D image of the data to remove scars from.
+        marked: np.ndarry
+            A 2-D image of pixels that stores the positions of scars marked for removal. The value of the pixel is a
+            floating point value that represents how strongly the algorithm considers it to be a scar.
+            This may or may not already contain non-zero values given that previous iterations of scar removal
+            may have been performed.
+        threshold_low: float
+            A value that when multiplied with the standard deviation, acts as a threshold to determine if an increase
+            or decrease in height might constitute the top or bottom of a scar.
+        max_scar_width: int
+            A value that dictates the maximum width that a scar can be. Note that this does not mean horizontal width,
+            rather vertical, this is because we conside scars to be laying flat, horizontally, so their width is
+            vertical and their length is horizontal.
+
+        Returns
+        -------
+        None
+        """
+
+        # Unpack row, col
+        row = row_col[0]
+        col = row_col[1]
+
+        # If sharp enough rise
+        min_scar_value = img[row + 1, col]
+        max_border_value = img[row, col]
+        if min_scar_value - max_border_value > threshold_low * stddev:
+            # Possibly in a scar
+            for k in range(1, max_scar_width + 1):
+                if row + k + 1 >= img.shape[0]:
+                    # Bottom of image, break
+                    break
+                min_scar_value = min(min_scar_value, img[row + k, col])
+                max_border_value = max(img[row, col], img[row + k + 1, col])
+                # Check if scar ended
+                if min_scar_value - max_border_value > threshold_low * stddev:
+                    while k:
+                        val = (img[row + k, col] - max_border_value) / stddev
+                        marked[row + k, col] = val
+                        k -= 1
+
+    def mark_if_negative_scar(
```

Comment:
I'm surprised this and other functions don't raise a PyLint [no-self-use](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-self-use.html) error. :thinking:

Reply:
Made `mark_if_negative_scar` and `mark_if_positive_scar` static.